controlplane/kubeadm/api/v1beta1/conversion.go

```diff
@@ -77,6 +77,12 @@ func Convert_v1beta2_KubeadmControlPlaneStatus_To_v1beta1_KubeadmControlPlaneSta
 		out.UnavailableReplicas = in.Deprecated.V1Beta1.UnavailableReplicas
 	}
 
+	// Move initialization to old fields
+	if in.Initialization != nil {
+		out.Initialized = in.Initialization.ControlPlaneInitialized
+		out.Ready = in.Initialization.ControlPlaneInitialized
```

Comment:
As discussed, maybe we should set `out.Ready` via `out.Ready = in.Deprecated.V1Beta1.ReadyReplicas > 0`